treeherder/model/derived/jobs.py

```diff
@@ -656,6 +672,9 @@ def store_job_data(self, data):
                 debug_show=self.DEBUG,
                 placeholders=coalesced_job_guid_placeholders,
                 executemany=True)
+            for (coalesced_to_guid, job_guid) in coalesced_job_guid_placeholders:
```

Comment:
This doesn't actually instantiate a new job object, the filter/update gets reduced down to an sql update statement after Django has had its way with it. I'm not particularly worried about this code being inefficient vs. everything else in this jobs model. My feeling is that we should leave it as-is, and let someone else do the profiling of this code if/when it becomes a problem. If history is anything to go by, the bottlenecks will not be where we think they are.